### ISSUE TITLE
Fail to reproduce results in Figure 4

### ISSUE THREAD

**Issue by PhoenixGS** [EvolvingLMMs-Lab/LLaVA-OneVision-1.5#91]
Thank you so much for open-sourcing such a great multimodal training model and framework! We have made some attempts based on your work, but we are currently encountering some difficulties.

We are reproducing the 0M results in [Figure 4](https://arxiv.org/abs/2509.23661) while the results are signficantly worse than reported. Our practice is listed in the following and looking forward to your help!

### Evaluation Setup

We use the [VLMEvalKit](https://github.com/open-compass/VLMEvalKit) framework for evaluation. The environment includes torch 2.9.0, transformers 4.57.1, and NVIDIA A800-SXM4-80GB GPUs. The belowing results show that our evluation is alinged with the paper.

| Source | MathVista | MMMU_val | MMStar | OCRBench | Avg of 4 |
| :--- | :---: | :---: | :---: | :---: | :---: |
| Official Report | 67.9 | 52.7 | 64.9 | 80.0 | 66.4 |
| Our evaluation | 67.6 | 54.3 | 64.1 | 80.8 | 66.7 |

### Training

We followed the setup in [Stage 1 Alignment-Training](https://github.com/EvolvingLMMs-Lab/LLaVA-OneVision-1.5/blob/58fa0624d212ad4afa673fcbc9b34c1d53800feb/examples/llava_ov_1_5/quick_start/stage_1_alignment_llava_ov_4b.sh#L4) and [Stage 2 Instruct-Training](https://github.com/EvolvingLMMs-Lab/LLaVA-OneVision-1.5/blob/58fa0624d212ad4afa673fcbc9b34c1d53800feb/examples/llava_ov_1_5/quick_start/stage_2_instruct_llava_ov_4b.sh#L4).

Base Model: [LLaVA-OneVision-1.5-4B-stage0](https://huggingface.co/lmms-lab/LLaVA-OneVision-1.5-4B-stage0)
Data: LLaVA-Pretrain-558K for pre-training and LLaVA-NeXT-780K for instruction tuning.

| Stage | Global Batch Size | Learning Rate | Warmup | Max Steps | Max Length | Freezing |
| :--- | :---: | :---: | :---: | :---: | :---: | :--- |
| **Pre-training (Stage 1)** | 224 | 1e-4 | ratio 0.002 | 2500 | 2048 | Freeze LLM & ViT |
| **SFT (Stage 2)** | 224 | 1e-5 | ratio 0.002 | 3500 | 32768 | Full Training |

### Results

Our results are significantly worse than results in Figure 4 of the paper.

| Setting | MMMU_val | MMStar | MMBench EN |
| -- | -- | -- | -- |
| Figure 4  | ~44.0 | ~47.0 | ~75.7 |
| Our reproduced  | 39.8 (-4.2) | 44.7 (-2.3) | 64.8 (-10.9) |

### Question

Could you provide the detail reproducing guidance for results in Figure 4? Your help will be greatly appreciated!

**Comment by yiyexy** [comment]
Could you provide the loss log for stage 1?

**Comment by PhoenixGS** [comment]
Hi @yiyexy, thank you for the quick response!

I have attached the training loss curve and the raw training log for Stage 1 below.

Please let me know if you need any other specific metrics.

<img width="680" height="550" alt="Image" src="https://github.com/user-attachments/assets/f0bbf125-e125-47e3-95d1-29fa0a4bb738" />

[output.log](https://github.com/user-attachments/files/24444934/output.log)

**Comment by yiyexy** [comment]
@PhoenixGS Generally, the loss in stage 1 can be reduced to around 2.x. Did your loss reach this level? By the way, can you use lmms-eval to evaluate your model?

**Comment by PhoenixGS** [comment]
@yiyexy Hello, this [link](https://wandb.ai/THUNLP-MM/LLaVA-OneVision-1.5/reports/LLaVA-OneVision-1-5--VmlldzoxNTU0NDQyNg?accessToken=acuyx20061901qyls4tmhh1y69x9f40bvj1phc8454b3msf4nbrwi14qor2na93q) provides detailed logs for our stage-1 and stage-2 training. In stage-1, the training loss eventually decreased to around 2.7.
We are temporarily using the vlmevalkit framework for evaluation, aligned with the settings in lmms-eval.
We want to know if there are any misalignments in the training data or training parameter settings. Thank you very much for your reply!

**Comment by yiyexy** [comment]
@PhoenixGS I understand. I’m waiting for your new results using lmm-eval. Your loss looks normal, and we’re using the same hyperparameters as those in the open repo.

**Comment by PhoenixGS** [comment]
H @yiyexy ,
I wanted to share my latest evaluation results. I've conducted additional experiments and here's what I found:

Configuration | MMMU_val | MMStar | MMBench
-- | -- | -- | --
2.5K PT + 0M mid-train + LLaVA-NExT SFT (vlmevalkit method) | 39.8 | 44.7 | 64.8
Target (from paper Figure 4) | ~44.5 | ~47.0 | ~76
2.5K PT + 0M mid-train + LLaVA-NExT SFT (lmms-eval method) | 39.8 | 44.9 | 71.0

We observed that there's a notable difference in MMBench scores between vlmevalkit (64.8) and lmms-eval (71.0), suggesting that evaluation methodology plays a significant role. However, even with the lmms-eval results, there's still a gap compared to the target metrics. Therefore, we believe there may still be some mismatches, and hope to receive some guidance.

Any help would be greatly appreciated!

**Comment by yiyexy** [comment]
Could you please provide your training repository? I will also look up my previous training log and share it with you.

**Comment by PhoenixGS** [comment]
Thank you so much for your reply! We are currently preparing our training framework. Could you please provide your training logs so we can make a comparison beforehand?
In addition, we would like to know if the 4M mid-training data used in Figure 4 of the paper and the 3M data used in the quick start were both sampled from the 85M dataset of mid-training with the same distribution?
